vdgsa_backend/rental_viols/views/user.py

```diff
@@ -0,0 +1,114 @@
+    template_name = 'users/move.html'
+
+    def get(self, request: HttpRequest, *args: Any, **kwargs: Any):
+        context = {}
+        context['user'] = User.objects.get(pk=self.kwargs['pk'])
+        return render(
+            self.request,
+            'users/move.html', context
+        )
+
+    def post(self, request: HttpRequest, *args: Any, **kwargs: Any):
+        oldUserId = self.kwargs['pk']
+        newUserId = self.request.POST.get('user_id')
+        if (newUserId and oldUserId):
+            cursor = connection.cursor()
+            cursor.execute('UPDATE rental_viols_rentalhistory SET renter_num_id = %(newUserId)s \
+                                WHERE renter_num_id = %(oldUserId)s; \
+                            UPDATE rental_viols_waitinglist SET renter_num_id = %(newUserId)s \
+                                WHERE renter_num_id = %(oldUserId)s; \
+                            UPDATE rental_viols_viol SET renter_id = %(newUserId)s \
+                                WHERE renter_id = %(oldUserId)s; \
+                            UPDATE rental_viols_viol SET storer_id = %(newUserId)s \
+                                WHERE storer_id = %(oldUserId)s; \
+                            UPDATE rental_viols_bow SET storer_id = %(newUserId)s \
+                                WHERE storer_id = %(oldUserId)s; \
+                            UPDATE rental_viols_case SET storer_id = %(newUserId)s \
+                                WHERE storer_id = %(oldUserId)s; ', {'newUserId': newUserId,
+                                                                     'oldUserId': oldUserId})
+
+        return redirect(reverse('renter-info', args=[newUserId]))
+
+
+class ViewUserInfo(RentalViewBase, DetailView):
+    model = User
+    template_name = 'renters/userDetail.html'
+
+    def get_context_data(self, **kwargs):
+
+        context = super(ViewUserInfo, self).get_context_data(**kwargs)
+        context['history'] = (RentalHistory.objects.all().filter(
+                              Q(event=RentalEvent.rented)
+                              | Q(event=RentalEvent.renewed)
+                              | Q(event=RentalEvent.returned)).filter(
+                                  renter_num=self.kwargs['pk'])).annotate(
+                                      rental_end_date=Max('rental_end'))
+        # .annotate(rental_end_date=Max('rental_end')
+        return context
+
+
+class UserSearchView(RentalViewBase, View):
+    """Filter list of users """
+    @csrf_exempt
+    def get(self, request: HttpRequest, *args: Any, **kwargs: Any):
+        url_parameter = request.GET.get("q")
+        context = {}
+        if url_parameter:
+            users = User.objects.filter(Q(first_name__icontains=url_parameter)
+                                        | Q(last_name__icontains=url_parameter))
+        else:
+            users = User.objects.all()
+
+        context["users"] = users
+        if request.is_ajax():
```

Comment:
I think request.is_ajax() is deprecated as of Django 3.2. It may be cleaner to just pull this out into a separate view since it's serving different data.

Reply:
Turns out it's not needed at all - This view is always ajax - i.e. return json